src/main/java/org/generationcp/middleware/dao/GermplasmDAO.java

```diff
@@ -1061,12 +1062,59 @@ public Germplasm getByGIDWithMethodType(final Integer gid) {
 		}
 	}
 
+	public List<Germplasm> getByGIDListWithMethodAndBibref(final Set<Integer> gids) {
+
+		if (gids.isEmpty()) {
+			return new ArrayList<>();
+		}
+
+		try {
+			final Criteria criteria = this.getSession().createCriteria(Germplasm.class);
+			criteria.createAlias("method", "method", CriteriaSpecification.LEFT_JOIN);
+			criteria.createAlias("bibref", "bibref", CriteriaSpecification.LEFT_JOIN);
+			criteria.add(Restrictions.in("gid", gids));
+			criteria.add(Restrictions.eq("deleted", false));
+			criteria.add(Restrictions.eq("grplce", 0));
+
+			return criteria.list();
+
+		} catch (final HibernateException e) {
+			final String message =
+				"Error with getByGIDListWithMethodAndBibref(gids=" + gids.toString() + GermplasmDAO.QUERY_FROM_GERMPLASM + e.getMessage();
+			GermplasmDAO.LOG.error(message, e);
+			throw new MiddlewareQueryException(message, e);
+		}
+	}
+
+	public List<Germplasm> getByUUIDListWithMethodAndBibref(final Set<String> uuids) {
```

Comment:
These 2 methods can be merged to one as `getByGIDsOrGUUIDs(final Set<Integer> gids, final Set<String> guuids)`